src/components/ec/base/ucc_ec_base.h

```diff
@@ -47,6 +47,62 @@ typedef struct ucc_ec_ops {
     ucc_status_t (*event_test)(void *event);
 } ucc_ec_ops_t;
 
+typedef struct ucc_ee_executor {
+    ucc_ee_type_t  ee_type;
+    void          *ee_context;
+} ucc_ee_executor_t;
+
+typedef enum ucc_ee_executor_task_type {
+    UCC_EE_EXECUTOR_TASK_TYPE_WAIT         = UCC_BIT(0),
+    UCC_EE_EXECUTOR_TASK_TYPE_REDUCE       = UCC_BIT(1),
+    UCC_EE_EXECUTOR_TASK_TYPE_REDUCE_MULTI = UCC_BIT(2),
+    UCC_EE_EXECUTOR_TASK_TYPE_COPY         = UCC_BIT(3),
+} ucc_ee_executor_task_type_t;
+
+typedef struct ucc_ee_executor_params {
+    uint64_t        mask;
+    ucc_ee_type_t   ee_type;
+    uint64_t        task_types;
+} ucc_ee_executor_params_t;
+
+#define UCC_EE_EXECUTOR_NUM_BUFS 9
```

Comment:
> Do we need some interface in EC, so that TL could query that interface to get max_reduce_buffers

added

>  If we have > 8 GPUs will we have to extend this #define ?

yes, or handle this case in algorithm